lib/spaceship/tunes/tunes_client.rb

```diff
@@ -776,19 +791,38 @@ def remove_tester_from_app!(tester, app_id)
 
     private
 
+    def with_tunes_retry(tries = 3, &block)
+      return block.call
+    rescue Spaceship::TunesClient::ITunesConnectTemporaryError => ex
+      unless (tries -= 1).zero?
+        msg = "ITC temporary save error received: '#{ex.message}'. Retrying after 60 seconds (remaining: #{tries})..."
```

Comment:
Yes we could. Right now I am aiming for resilience, then fine tuning. In the end I would love to move some of this into either a faraday middleware, or using a custom more declarative "retrier".

For example, out of the last 5 runs, they all ended up failing here. I increased it to 60 seconds and it still fails sometimes.

Here are the last 2, with a 60 second sleep in between retries:
```
ITC temporary save error received: 'We're temporarily unable to save your changes. Please try again later.'. Retrying after 60 seconds (remaining: 1)...
ESC[37m[19:47:24]: ESC[0mFailed / completed after 5 hour(s) 16 min(s) 32 sec(s)
/Users/lacostej/.rvm/gems/ruby-2.2.4/gems/spaceship-0.19.3/lib/spaceship/tunes/tunes_client.rb:218:in `handle_itc_response': We're temporarily unable to save your changes. Please try again later. (Spaceship::TunesClient::ITunesConnectTemporaryError)
        from /Users/lacostej/.rvm/gems/ruby-2.2.4/gems/spaceship-0.19.3/lib/spaceship/tunes/tunes_client.rb:361:in `block in update_app_version!'
```


```
ITC temporary save error received: 'We're temporarily unable to save your changes. Please try again later.'. Retrying after 60 seconds (remaining: 1)...
ESC[37m[21:20:34]: ESC[0mFailed / completed after 1 hour(s) 33 min(s) 8 sec(s)
/Users/lacostej/.rvm/gems/ruby-2.2.4/gems/spaceship-0.19.3/lib/spaceship/tunes/tunes_client.rb:218:in `handle_itc_response': We're temporarily unable to save your changes. Please try again later. (Spaceship::TunesClient::ITunesConnectTemporaryError)
        from /Users/lacostej/.rvm/gems/ruby-2.2.4/gems/spaceship-0.19.3/lib/spaceship/tunes/tunes_client.rb:361:in `block in update_app_version!'
        from /Users/lacostej/.rvm/gems/ruby-2.2.4/gems/spaceship-0.19.3/lib/spaceship/tunes/tunes_client.rb:795:in `call'
```

In the one that lasted 5 h 30 min, I had
* 0 timeout / connection failed (they appear more seldomly now given the 300 seconds 
* 20+ auth lost
* 6 unable to save (last one fatal)
connection timeout)